.github/workflows/awesome_workflow.yml

```diff
@@ -100,24 +88,18 @@ jobs:
 
             with open("DIRECTORY.md", "w") as out_file:
                 out_file.write(build_directory_md(".") + "\n")
-      - name: Update DIRECTORY.md
+      - name: Commit DIRECTORY.md
+        run: git commit -m "updating DIRECTORY.md" DIRECTORY.md ||  true
+      - name: Get file changes
         run: |
-          cat DIRECTORY.md
-          git config --global user.name github-actions
-          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
-          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
-          git add DIRECTORY.md
-          git commit -am "updating DIRECTORY.md" ||  true
-          git push --force origin HEAD:$GITHUB_REF || true
-      - name: Install CPPLINT
-        run: |
-          python -m pip install cpplint
           git remote -v
           git branch
           git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
           git diff --diff-filter=dr --name-only origin/master > git_diff.txt
           echo "Files changed-- `cat git_diff.txt`"
-      - name: cpplint_modified_files
+      - name: Configure for static lint checks
+        run: cmake -B build -S . -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```

Comment:
compiling first gives `clang-tidy` access to all the header files and settings used to compile the programs

Reply:
Should some of these comments be added as comments in the .yml file so that it is self-documenting?

Reply:
excellent idea.